### PR TITLE
Lightning bolt weapon proc

### PR DESCRIPTION
**Pull request by Durhamite2118**
Similar to molten/glacial except lightning / electric damage based

### CODE REVIEW

File: code/code/spec/spec_objs_weapons.cc [hunk 1]
```diff
@@ -2748,3 +2748,197 @@ int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
   return FALSE;
 }
 
+// Summary of this weapon proc
```
Comment (cizra) [inline code/code/spec/spec_objs_weapons.cc:2751]: indent missing

File: code/code/spec/spec_objs_weapons.cc [hunk 2]
```diff
@@ -2748,3 +2748,197 @@ int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
...
+TThing *t;
+
+if ((t = vict->unequip(slot)))
+*(vict->roomp) += *t;
+
+for (wearSlotT j=MIN_WEAR; j < MAX_WEAR; j++) {
+if (!vict->hasPart(j))
+continue;
+
+if (!vict->limbConnections(j)) {
```
Comment (cizra) [inline code/code/spec/spec_objs_weapons.cc:2860]: Is this file mostly copy-paste? Maybe we could consolidate the code with bonebreak or lightsaber or such, passing in the specific flavor strings and chances and whatnot?

**Review by cizra — CHANGES_REQUESTED** [review]

File: code/code/spec/spec_objs_weapons.cc [hunk 3]
```diff
@@ -2748,3 +2748,196 @@ int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
...
+  buf = format("$n's $p flashes with a blinding white light and emits a bolt of lightning!");
+
+  act(buf, FALSE, ch, o, vict, TO_NOTVICT, ANSI_WHITE);
+  buf = format("$n's $p flashes with a blinding white light and emits a bolt of lightning!");
+
+  act(buf, FALSE, ch, o, vict, TO_VICT, ANSI_WHITE);
+
+  buf = format("You look down in horror and disbelief, as $n's $p turns your %s into ash!") % limb;
+
+  act(buf, FALSE, ch, o, vict, TO_VICT, ANSI_GREY);
```
Comment (wershlak) [inline code/code/spec/spec_objs_weapons.cc:2887]: https://github.com/sneezymud/sneezymud/blob/master/code/code/sys/ansi.h#L93-L114

There's no ANSI_GREY

Simple mistake but I think you need to be able to build and test your changes locally if you want to be able to contribute. I usually would compile a thing like this change and ensure the proc fires and does what I think it should, doesn't crash the mud and all the ansi looks pretty. Once, you get a local instance running it's not too much work to run a short test on each change.

@cizra's instructions here worked great: https://github.com/sneezymud/sneezymud-docker#to-develop-using-a-clean-database

File: code/code/spec/spec_objs.cc [hunk 4]
```diff
@@ -6784,7 +6784,7 @@ extern int shadowWeapon(TBeing *vict, cmdTypeT cmd, const char *, TObj *o, TObj
 extern int livingVines(TBeing *vict, cmdTypeT cmd, const char *, TObj *o, TObj *);
 extern int dkSword(TBeing *vict, cmdTypeT cmd, const char *, TObj *o, TObj *);
 extern int moltenWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *);
-extern int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *);
```
Comment (wershlak) [inline code/code/spec/spec_objs.cc:6787]: Why'd you delete this? That's not right

File: code/code/spec/spec_objs_weapons.cc [hunk 5]
```diff
@@ -2748,3 +2748,196 @@ int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
...
+      found |= (vict->isLimbFlags(slot, PART_MISSING));
+
+      ok = true;
+    }
+
+  if (found || !ok) 
+  return false;
+
+// Accounting for electricity immunity since this is electricity-based wither limb
+  if (vict->isImmune(IMMUNE_ELECTRIC, slot))
```
Comment (wershlak) [inline code/code/spec/spec_objs_weapons.cc:2819]: IMMUNE_ELECTRIC isn't a thing either.

**Review by wershlak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**